src/features/messages/cards/TransactionCard.tsx

```diff
@@ -236,11 +242,64 @@ function CallDataModal({ debugResult }: { debugResult?: MessageDebugResult }) {
           </p>
           <LabelAndCodeBlock label="Recipient contract address:" value={contract} />
           <LabelAndCodeBlock label="Handle function input calldata:" value={handleCalldata} />
+          <button onClick={async()=>{
+            handleClick()
+            await simulateCall({contract,handleCalldata,chainId})
+          }}
+            disabled={disabled}
+            className='underline text-blue-400'
+          >
+            {buttonText}
+          </button>
+
         </div>
       </Modal>
     </>
   );
 }
+const simulateCall=async({contract,handleCalldata,chainId}:{contract:string,handleCalldata:string,chainId:ChainId})=>{
+  const TENDERLY_USER=''
+  const TENDERLY_PROJECT=''
+  const TENDERLY_ACCESS_KEY=''
+
+  console.time('Simulation')
+
+  const resp = await axios.post(
+    `https://api.tenderly.co/api/v1/account/${TENDERLY_USER}/project/${TENDERLY_PROJECT}/simulate`,
+    {
+      save: true,
+      save_if_fails: true, 
+      simulation_type: 'full',
+      network_id: chainId, 
+      from: '0xdc6bdc37b2714ee601734cf55a05625c9e512461',//can be any address, doesn't matter
+      to: contract,
+      input:handleCalldata,
+      gas: 8000000,
+      gas_price: 0,
+      value: 0,
+    },
+    {
+      headers: {
+        'X-Access-Key': TENDERLY_ACCESS_KEY as string,
+      },
+    }
+  );
+  console.timeEnd('Simulation');
+  console.log(resp.data)
```

Comment:
Use the `logger` utility instead of console and add a message to provide context for the data you're logging